### PR TITLE
DEV-8137 Add multi-currency transactions aggregation tutorial

### PR DESCRIPTION
**Pull request by kkeenan-finbourne**
- Add an example in the Valuation tutorial of a portfolio in
a base currency having transactions applied in instruments of
different currencies

# Pull Request Checklist

- [ ] Read the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] Tests pass
- [ ] Raised the PR against the `develop` branch

# Description of the PR

Describe the code changes for the reviewers, explain the solution you have provided and how it fixes the issue

### CODE REVIEW

File: sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java [hunk 1]
```diff
@@ -83,6 +87,86 @@ public void run_aggregation_with_buy() throws ApiException
...
+        String quotesScope = uuid;
+
+        // begin test data preparation...
+
+        //  prepare portfolio in baseline GBP currency
+        String portfolioId = String.format("Id-%s", uuid);
+        String portfolioDisplayName = String.format("Portfolio-%s", uuid);
+        CreateTransactionPortfolioRequest transactionPortfolioRequest = createTransactionPortfolioRequest(portfolioId, portfolioDisplayName, "GBP");
+
+        //  prepare transactions  across GBP and USD. transactions are cash funding and stock fundings
```
Comment (rizwansaeed) [inline sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java:103]: these are transactions are both transfer in of both cash and stock, perhaps rephrase the comment as:

"prepare transactions across GBP and USD, transactions are transfer in of cash and stock"

File: sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java [hunk 2]
```diff
@@ -178,8 +262,136 @@ private void run_aggregation(
...
+     * @param baseCcy - base currency of the quote
+     * @param counterCcy - counter currency of the quote
+     * @param rate - the exchange rate
+     * @return
+     */
+    private UpsertQuoteRequest createFxUpsertQuoteRequest(String baseCcy, String counterCcy, double rate){
+        return new UpsertQuoteRequest()
+                .quoteId(new QuoteId()
+                        .quoteSeriesId(new QuoteSeriesId()
+                                .provider("DataScope")
```
Comment (rizwansaeed) [inline sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java:374]: can you change this to `Lusid`, `DataScope` is a specific vendor provider

File: sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java [hunk 3]
```diff
@@ -178,8 +262,136 @@ private void run_aggregation(
...
+                                .instrumentId(baseCcy + "/" + counterCcy)
```
Comment (rizwansaeed) [inline sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java:375]: @slemasne is this a specific/required format? 

@kkeenan-finbourne we should add a comment explaining so if it is

File: sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java [hunk 4]
```diff
@@ -178,8 +262,136 @@ private void run_aggregation(
...
+                        .effectiveAt(EFFECTIVE_DATE.toString())
+                )
+                .metricValue(new MetricValue().value(value).unit(currency));
+    }
+
+    /**
+     * Creates an fx entry quote request.
+     *
+     * @param baseCcy - base currency of the quote
+     * @param counterCcy - counter currency of the quote
```
Comment (rizwansaeed) [inline sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java:366]: "foreign" is the more common term when talking about currency pairs

File: sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java (hunk 3 above)
Comment (kkeenan-finbourne) [reply sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java:375]: I think it is a requirement to follow that format. Amending it to say "USD-GBP" for example results in a market data resolve failure in the aggregation engine with it searching for USD/GBP. Will update with relevant comment

File: sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java (hunk 1 above)
Comment (kkeenan-finbourne) [reply sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java]: Done

File: sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java (hunk 2 above)
Comment (kkeenan-finbourne) [reply sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java]: Done. To do so I've also had to add LUSID as the FX market context supplier in the configuration recipe. Added relevant comment.

File: sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java (hunk 4 above)
Comment (kkeenan-finbourne) [reply sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java]: Done

File: sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java [hunk 5]
```diff
@@ -281,7 +281,12 @@ private void run_aggregation(
         ConfigurationRecipe configurationRecipe = new ConfigurationRecipe()
                 .code("quotes_recipe")
                 .market(new MarketContext()
-                        .suppliers(new MarketContextSuppliers().equity(MarketContextSuppliers.EquityEnum.DATASCOPE))
+                        // equity data and fx data are sourced from different data providers, reference the createUpsertQuote... methods
+                        // to view how the quote requests are mapped to providers.
+                        .suppliers(new MarketContextSuppliers()
+                                .equity(MarketContextSuppliers.EquityEnum.DATASCOPE)
```
Comment (rizwansaeed) [inline sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java:287]: my fault I probably wasn't clear in my last comment, could we change DataScope to LUSID unless we intentionally need to show 2 different providers?

Comment (kkeenan-finbourne) [reply sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java]: Done. All test cases now only use Lusid data provider.

**Pull request merged**